core/types/celo_transaction_marshalling.go

```diff
@@ -37,6 +37,7 @@ func celoTransactionMarshal(tx *Transaction) ([]byte, bool, error) {
 		enc.Nonce = (*hexutil.Uint64)(&itx.Nonce)
 		enc.To = tx.To()
 		enc.Gas = (*hexutil.Uint64)(&itx.Gas)
+		enc.GasPrice = (*hexutil.Big)(itx.GasPrice)
```

Comment:
I found one thing while I was writing a test:

Originally Celo's LegacyTx set `null` to `gasPrice` field on json marshaling. But Celo LegacyTx takes `gasPrice` field from JSON on unmarshaling. It causes mismatch after marshaling and unmarshaling. 

Is this fix right? If no, I will remove this new line and do small fixes of unit tests